_data/heroku-landing-page-tabs-code.yml

```diff
@@ -0,0 +1,111 @@
+          launch_type                    = "FARGATE"
+          platform_version               = "1.4.0"
+
+          # Network information is necessary for Fargate, as it required VPC type
+          ecs_task_definition_network_mode = "awsvpc"
+          ecs_service_network_configuration = {
+            subnets          = data.aws_subnet_ids.default.ids
+            security_groups  = [aws_security_group.ecs_task_security_group.id]
+            assign_public_ip = true
+          }
+
+          # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-tasks-size.
+          # Specify memory in MB
+          task_cpu    = 256
+          task_memory = 512
+
+          enable_ecs_deployment_check      = var.enable_ecs_deployment_check
+          deployment_check_timeout_seconds = var.deployment_check_timeout_seconds
+
+          # EFS volumes
+          efs_volumes = local.efs_volumes
+
+          # Make sure all the ECS cluster resources are deployed before deploying any ECS service resources. This is also
+          # necessary to avoid issues on 'destroy'.
+          depends_on = [aws_ecs_cluster.fargate_cluster]
+        }
+      </code>
+      </pre>
+      <figure>
+- id: ecs
+  title: ECS
+  content: |
+    <figure><figcaption>ecs-cluster/main.tf</figcaption>
+      <pre>
+      <code class="language-js">
+        module "ecs_cluster" {
+          source = "git::git@github.com:gruntwork-io/terraform-aws-ecs.git//modules/ecs-cluster?ref=v1.0.0"
+
+          cluster_name = var.ecs_cluster_name
+
+          # Make the max size twice the min size to allow for rolling out updates to the cluster without downtime
+          cluster_min_size = 2
+          cluster_max_size = 4
+
+          cluster_instance_ami              = var.ecs_cluster_instance_ami
+          cluster_instance_type             = module.instance_type.recommended_instance_type
+          cluster_instance_keypair_name     = var.ecs_cluster_instance_keypair_name
+          cluster_instance_user_data        = data.template_file.user_data.rendered
+          enable_cluster_container_insights = true
+
+          vpc_id         = var.vpc_id
+          vpc_subnet_ids = var.ecs_cluster_vpc_subnet_ids
+
+          alb_security_group_ids = [module.alb.alb_security_group_id]
+
+          custom_tags_security_group = {
+            Foo = "Bar"
+          }
+
+          custom_tags_ec2_instances = [
+            {
+              key                 = "Foo"
+              value               = "Bar"
+              propagate_at_launch = true
+            },
+          ]
+        }
+      </code>
+      </pre>
+      <figure>
+- id: eks
+  title: EKS
+  content: |
+    <figure><figcaption>eks-cluster/main.tf</figcaption>
+      <pre>
+      <code class="language-js">
+        module "eks_cluster" {
+          source = "git::git@github.com:gruntwork-io/terraform-aws-eks.git//modules/eks-cluster-control-plane?ref=v0.1.0"
```

Comment:
Same here: we should use the Service Catalog. Here's a snippet:

```hcl
module "k8s_service" {
  source = "git::git@github.com:gruntwork-io/terraform-aws-service-catalog.git//modules/services/k8s-service?ref=v0.51.0"  

  application_name       = "my-heroku-app"
  namespace              = "my-namespace"
  desired_number_of_pods = 3

  container_port  = 5000
  container_image = {
    repository = "https://123456789012.dkr.ecr.us-east-1.amazonaws.com/my-heroku-app"
    tag        = "v1"
  }

  domain_name = "example.your-domain.com"
}

module "eks_cluster" {
  source = "git::git@github.com:gruntwork-io/terraform-aws-service-catalog.git//modules/services/eks-cluster?ref=v0.51.0"  

  cluster_name          = "my-eks-cluster"
  kubernetes_version    = "1.20"
  cluster_instance_ami  = "ami-abcd1234"

  managed_node_group_configurations = {
    default = {
      min_size       = 3
      max_size       = 10
      instance_types = ["m4.large"]
    }
  }
}

provider "kubernetes" {
  host                   = module.eks_cluster.endpoint
  cluster_ca_certificate = module.eks_cluster.certificate_authority.0.data
  token                  = module.eks_cluster.kubernetes_token[0].token
}
```

Reply:
funny! I actually had the service catalog at first.... and then decided to swap them out! Good one :)